### ISSUE TITLE
Why Yolov10 performance on custom dataset worse than yolov7. What hyperparameters can be adjusted to improve performance?

### ISSUE THREAD

**Issue by mazatov** [THU-MIG/yolov10#401]
I'm training on my custom object detection dataset. I'm detecting a ball in sports images. So I have one object that is quite small!

I trained on the same model Yolov10-B and default Yolov7. The performance of Yolov10 was quite a bit worse, which surprised me. My guess is some hyperparameters might need adjusting. One thing I noticed was that Yolov10 exited after 189 epochs because no improvement was observed in 100 epochs while Yolov7 trained for all 300 epochs.

What hyperparameters can be adjusted to improve performance?

**Comment by HumamHelfawi** [comment]
I have the same problem. YoloV8 perform both better in term of accuracy and performance than YoloV10

**Comment by happyiminjay1** [comment]
I get worse result compared to the yolov5 on my custom dataset which has only 5 classes. Yolov5 is much faster.. Does the class number matters on the performance?

**Comment by CFZ1** [comment]
I have the same problem. Why did YOLOV10 experience a situation where the performance of subsequent epochs could no longer be improved? thank you.

**Comment by CFZ1** [comment]
> I have the same problem. Why did YOLOV10 experience a situation where the performance of subsequent epochs could no longer be improved? thank you.
自己来回复一下，我出现的问题在于，我一开始为了简单一点，做了几处改动，（1）不使用复杂的数据增强（只使用图像翻转+调整图像亮度），（2）所有参数使用同样的decay。（3）使用20epoch在自己的数据集上。但是怎么样调整，模型似乎到epoch=5之后都上不去了。
现在使用默认的数据增强。优化器为：AdamW(lr=0.000714, momentum=0.9) with parameter groups weight(decay=0.0005)。模型性能上来了。可能在我的数据集上，重要的参数有：数据增强、lr、decay、reg_max（在head.py中）、以及momentum（momentum可能是这里最不重要的）
reg_max：因为我的输入图像是1024×1024，reg_max决定了预测边框的上限，边框的上限=2×(reg_max-1)×max(特征图stride)=2*15*32=960，在我的图像上调大一点会比较好。

**Comment by CFZ1** [comment]
> I have the same problem. Why did YOLOV10 experience a situation where the performance of subsequent epochs could no longer be improved? thank you.

自己来回复一下，我出现的问题在于，我一开始为了简单一点，做了几处改动，（1）不使用复杂的数据增强（只使用图像翻转+调整图像亮度），（2）所有参数使用同样的decay。（3）使用20epoch在自己的数据集上。但是怎么样调整，模型似乎到epoch=5之后都上不去了。
现在使用默认的数据增强。优化器为：AdamW(lr=0.000714, momentum=0.9) with parameter groups weight(decay=0.0005)。模型性能上来了。可能在我的数据集上，重要的参数有：数据增强、lr、decay、reg_max（在head.py中）、以及momentum（momentum可能是这里最不重要的）
reg_max：因为我的输入图像是1024×1024，reg_max决定了预测边框的上限，边框的上限=2×(reg_max-1)×max(特征图stride)=21532=960，在我的图像上调大一点会比较好。

**Comment by CFZ1** [comment]
> I have the same problem. Why did YOLOV10 experience a situation where the performance of subsequent epochs could no longer be improved? thank you.

自己来回复一下，我出现的问题在于，我一开始为了简单一点，做了几处改动，（1）不使用复杂的数据增强（只使用图像翻转+调整图像亮度），（2）所有参数使用同样的decay。（3）使用20epoch在自己的数据集上。但是怎么样调整（包括很多defalut.py中的参数，以及匹配策略的参数tal_topk、alpha、beta，以及reg_max），模型似乎到epoch=5之后都上不去了。
现在使用默认的数据增强。优化器为：AdamW(lr=0.000714, momentum=0.9) with parameter groups weight(decay=0.0005)。模型性能上来了。可能在我的数据集上，重要的参数有：数据增强、lr、decay、reg_max（在head.py中）、以及momentum（momentum可能是这里最不重要的）
reg_max：因为我的输入图像是1024×1024，reg_max决定了预测边框的上限，边框的上限=2×(reg_max-1)×max(特征图stride)=21532=960，在我的图像上调大一点会比较好。
